Sources/HipMobileUI/ViewModels/Views/ExhibitDetails/AppetizerViewModel.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (C) 2017 History in Paderborn App - Universität Paderborn
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//       http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.IO;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Models;
+using Xamarin.Forms;
+using System;
+using System.Windows.Input;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.DtoToModelConverters;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Managers;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.Common;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.Navigation;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.ViewModels.Pages;
+
+namespace PaderbornUniversity.SILab.Hip.Mobile.UI.ViewModels.Views.ExhibitDetails
+{
+    public class AppetizerViewModel : ExhibitSubviewViewModel, IDownloadableListItemViewModel
+    {
+        private Exhibit exhibit;
+        private ImageSource image;
+        private string text;
+        private string headline;
+
+        public AppetizerViewModel (Exhibit exhibit)
+        {
+            Exhibit = exhibit;
+            Headline = exhibit.Name;
+            Text = exhibit.Name;
+
+            // workaround for realm bug
+            var imageData = exhibit.Image.Data;
+            if (imageData != null)
+                {
+                Image = ImageSource.FromStream (() => new MemoryStream (imageData));
+                }
+            else
+                {
+                Image = ImageSource.FromStream (() => new MemoryStream (BackupData.BackupImageData));
+                }
+
+            IsDownloadButtonVisible = !Exhibit.DetailsDataLoaded;
+
+            DownloadCommand = new Command (OpenDownloadDialog);
+        }
+
+        private async void OpenDownloadDialog()
+        {
+            // Open the download dialog
+            downloadPage = new ExhibitRouteDownloadPageViewModel(Exhibit, this);
+            await Navigation.PushAsync(downloadPage);
+        }
+
+        public void CloseDownloadPage()
+        {
+            IoCManager.Resolve<INavigationService>().PopAsync();
```

Comment:
Does this actually work? `PopAsync` returns a `Task` and those normally need to be awaited.

Reply:
These changes are from some previous commit. I merged branch and got this